### ISSUE TITLE
Issues for using external fields in RZ

### ISSUE THREAD

**Issue by Yin-YinjianZhao** [ECP-WarpX/WarpX#1189]
[inputs.txt](https://github.com/ECP-WarpX/WarpX/files/4950268/inputs.txt)
When this input is used for RZ, the beam is found to diverge, however it should be focused.
![100](https://user-images.githubusercontent.com/56095356/87992009-256f9500-ca9c-11ea-8e5c-6d5558e19abd.png)
![300](https://user-images.githubusercontent.com/56095356/87992011-26082b80-ca9c-11ea-9a85-0f336b251663.png)
![500](https://user-images.githubusercontent.com/56095356/87992013-26082b80-ca9c-11ea-8f36-e34c37ff073e.png)
As shown by the three figures at 100, 300, and 500 time steps in r-z plots.

It is found that when the external fields are turned off:
```
#particles.E_ext_particle_init_style = "parse_E_ext_particle_function"
#my_constants.const = 9047564023865190.0 # me*c^2*kp^2/(2*e)
#particles.Ex_external_particle_function(x,y,z,t) = "const * x"
#particles.Ey_external_particle_function(x,y,z,t) = "const * y"
#particles.Ez_external_particle_function(x,y,z,t) = "0.0"
```
the beam becomes at 300, 1000, and 5000 time step:
![300 (1)](https://user-images.githubusercontent.com/56095356/87992129-74b5c580-ca9c-11ea-9882-cdc6b8243a25.png)
![1000](https://user-images.githubusercontent.com/56095356/87992132-754e5c00-ca9c-11ea-8ba8-db0665f30e16.png)
![5000](https://user-images.githubusercontent.com/56095356/87992134-754e5c00-ca9c-11ea-9c8c-f3c6e34d2ef7.png)

@RemiLehe  and I doubt there may be a bug related with the external fields in RZ, or we are not using it correct.

Helps are needed, thanks!

**Comment by Yin-YinjianZhao** [comment]
Some updates: @RemiLehe @dpgrote @RevathiJambunathan 

RZ and 3D are compared.

For 3D, with the input file [inputs_3d.txt](https://github.com/ECP-WarpX/WarpX/files/5019456/inputs_3d.txt), we have at t = 1e-11 s, where beam is red, ions are blue.

![3d](https://user-images.githubusercontent.com/56095356/89240377-cab45e00-d5b0-11ea-8af1-0162dec411b9.png)

For RZ, with the input file [inputs_rz.txt](https://github.com/ECP-WarpX/WarpX/files/5019460/inputs_rz.txt), we have at t = 1e-11 s, where beam is red, ions are blue.

![rz](https://user-images.githubusercontent.com/56095356/89240397-d142d580-d5b0-11ea-8aaf-da99c3d450eb.png)

For 3D, without external fields:

![3dnoE](https://user-images.githubusercontent.com/56095356/89240403-d738b680-d5b0-11ea-82ae-bd3e51e7ae43.png)

For RZ, without external fields:

![rznoE](https://user-images.githubusercontent.com/56095356/89240410-db64d400-d5b0-11ea-910b-4c20dac0169f.png)

So there seems to be no issue without external fields.

Besides, in RZ, 2\*2 ions are set per cell in 8\*8 cells. But along z, there seems to be only 1 ion per cell. ???

Jupyter notebook for plotting is 
[Visualization.txt](https://github.com/ECP-WarpX/WarpX/files/5019477/Visualization.txt)

**Comment by Yin-YinjianZhao** [comment]
Also, Reva has checked that the parser looks good by printing values from the code and they matched with those in the input file.

**Comment by RevathiJambunathan** [comment]
Hi @Yin-YinjianZhao Thank you for the update and for checking this. I wanted to mention that the number of ions per cell in the input_rz file should be 2 2 2.
Sorry! my bad. I gave you mis-information when we talked offline about only the first two dimensions being used. 
I checked this now, and in RZ, three values are provided for particles_per_cell. I tested this locally and I get 2 ions per cell in the z-direction upon adding that third value.
